pkg/azureclients/armclient/util.go

```diff
@@ -123,7 +123,11 @@ func DoHackRegionalRetryDecorator(c *Client) autorest.SendDecorator {
 			// Hack: retry the regional ARM endpoint in case of ARM traffic split and arm resource group replication is too slow
 			// Empty content and 2xx http status code are returned in this case.
 			// Issue: https://github.com/kubernetes-sigs/cloud-provider-azure/issues/1296
+			// It needs retry if ContentLength == -1, StatusCode 200 and empty body.
 			emptyResp := (response.ContentLength == 0 || trimmed == "" || trimmed == "{}") && response.StatusCode >= 200 && response.StatusCode < 300
+			if response.ContentLength == -1 && response.StatusCode >= 200 && response.StatusCode < 300 && (trimmed == "" || trimmed == "{}") {
```

Comment:
could we merge `response.ContentLength == -1` into L127, e.g. `response.ContentLength == 0 || response.ContentLength == -1 ...`

Reply:
Updated.